src/Stratis.Bitcoin/Builder/FullNodeBuilder.cs

```diff
@@ -161,6 +162,18 @@ public IFullNode Build()
 
             this.Services = this.BuildServices();
 
+            if (this.NodeSettings.PrintHelpAndExit)
+            {
+                foreach (var featureRegistration in this.Features.FeatureRegistrations)
+                {
+                    MethodInfo printHelp = featureRegistration.FeatureType.GetMethod("PrintHelp", BindingFlags.Public | BindingFlags.Static);
+
+                    printHelp?.Invoke(null, new object[] { this.NodeSettings.Network });
+                }
+
+                throw new NodeBuilderException("User requested to print help and exit");
```

Comment:
I don't really like the exception here.
Maybe check the value of printHelpAndExit  after the build() call. if false, do `await node.RunAsync();`, otherwise nothing.
That's the best idea I came up with, although I'm not a big fan.
Any other ideas?

Reply:
Maybe we can simply have the RunAsync exit without doing anything.